### DISCUSSION TITLE
How to model a submerged rigid object instead of making it as a floating object

### DISCUSSION THREAD

**Discussion by jialing95** [DualSPHysics/DualSPHysics#218] [DUALSPHYSICS_V5.2]
Hi ,folks,

I'm currently modeling a submerged object sliding down along the slope, but when I tried to give it density property, I could only set it as floating object, and apply Chrono collision method. It is not viable because the object should be heavier than water, and get sink to the bottom. 

Can anyone give me some suggestions? Thanks a lot!

**Comment by alexbexe** [comment]
Dear user, can you explain better the issue?
One suggestion is to impose mass if you know the actual value.

Regards
Alex

**Reply by jialing95** [reply]
Hi,  Alex

Fig. 1 is the general layout of my simulation.
![grilli_sketch](https://github.com/user-attachments/assets/3d417f52-0ba6-4162-9606-0740811f97cc)
Fig. 1  Model layout

I want the submerged 3D ellipse to slide down along the slope. But it remains still and looks like floating on the water, making small ripplpes (Fig. 2), so it occurs to me that something was wrong with its attribute.
![simu2](https://github.com/user-attachments/assets/482f7de8-e213-4d86-89b9-84a5cda52c16)
Fig. 2 Simulation result

This is how I did: 
When I set its density, I went to the panel in the FreeCAD by choosing it as floating object (Fig. 3). It's the only way so far I can do to add its density property. 
![3](https://github.com/user-attachments/assets/4dcc5150-d969-45ea-b0e0-319c346d76e6)
Fig. 3 Floating object with density attribute

To make it interact with slope and tank bottom, I applied Chrono as follows (note "gauss" is the object):
![Screenshot from 2024-07-23 17-52-36](https://github.com/user-attachments/assets/1677471d-25e1-4602-9752-03323ec68ef6)

I was wondering if there is another way to avoid it float while it can slide and interact with other soild.

Best regrads
Jialing

**Comment by alexbexe** [comment]
Imposing density but creating particles only in the faces of the figure, the final mass can not be the expected one.

So you have two options:
1) imposing density but filling the object with particles (autofill or similar)
2) imposing the actual value of mass (instead of density)

Regards

**Reply by jialing95** [reply]
Hi, Alex,

My current simulation uses option 1, but failed as mentioned above. 

Option 2 seems a new solution to me, because I thought mass or density will have no difference to rigid body. 

I will try it and get it back to you.

Thanks for help!
Jialing

**Comment by DualSPHysics** [comment]
Note that density and mass are related of course.
However using density, the mass is computed as the summation of individual masses of the particles created in the object.
Therefore you need to be sure that the number of particles and the volume occupied tend to the actual volume of the object.
SO you need to fill the object with particles for sure and the mass will converge to the actual value when you increase the resolution.

However if you impose the mass, the value is not computed by default and that mass value will be used in the simulation.

**Comment by jialing95** [comment]
Hi, I found the floating object stand still after I modified some parameters in the **xml file**. (To make the task easier, I lowered the water depth, so as to examine if it can detect collision when sliding down, unfortunately it failed.)
It looks like this now:
![Screenshot from 2024-07-26 15-31-28](https://github.com/user-attachments/assets/f27e17f3-155b-4c72-8668-5b873d1adb03)
The **PartFloating .vtk** seems to suggest it is integrated with the slope, but the **original stl file** as well as **Autual.vtk** show it has a certain distance bwteen the slope. 

The experiment that I'm duplicating is very sensitive to height magnitude, that's why I put the floating object at this height carefully. However, there must be something wrong with the inproporiate parameters in **_Def.xml** or the object stl itself.

I attach xml file here for you to review. 
The layout is like this, symmetric to x axis, 3 objects in total (slope, water tank, floating object):
![Screenshot from 2024-07-26 16-01-15](https://github.com/user-attachments/assets/c7397042-8e58-4cde-9438-158fcb510924)

[Grilli_61.zip](https://github.com/user-attachments/files/16392137/Grilli_61.zip)

Please let me know where the problem lies in, thanks so much!!
